app/Components/Sudoku Board/SudokuBoard.tsx

```diff
@@ -284,68 +289,85 @@ const SudokuBoard = (props: SudokuBoardProps) => {
   };
 
   const renderCell = (cell: CellProps, r: number, c: number) => {
-    // row and column values are incorrect here.
-    // console.log("RENDER CELL", c, "COLUMN", r, "ROW", cell)
-    const {
-      isHighlightIdenticalValues,
-      isHighlightBox,
-      isHighlightRow,
-      isHighlightColumn,
-    } = React.useContext(PreferencesContext);
-
-    let selected = sudokuBoard.selectedCell;
-    let isSelected = false;
-    let conflict = isConflict(r, c, cell);
-    let peer = false;
-    let box = false;
-    let row = false;
-    let column = false;
-    let sameValue = false;
-    if (selected != null) {
-      isSelected = c === selected.c && r === selected.r;
-      box = isCurrentCellAndSelectedCellInSameBox({ r: r, c: c }, selected);
-      row = isCurrentCellAndSelectedCellInSameRow({ r: r, c: c }, selected);
-      column = isCurrentCellAndSelectedCellInSameColumn(
-        { r: r, c: c },
-        selected
-      );
-      peer =
-        !conflict &&
-        ((box && isHighlightBox) ||
-          (row && isHighlightRow) ||
-          (column && isHighlightColumn));
-
-      const currentSelectedCell: CellProps | null = getCurrentSelectedCell();
-      if (currentSelectedCell == null) {
-        return;
-      }
-      const selectedEntry = currentSelectedCell.entry;
-      let currentEntry = cell.entry;
-      sameValue =
-        !conflict &&
-        isHighlightIdenticalValues &&
-        selectedEntry === currentEntry &&
-        currentEntry != 0;
+    const selectedCell = sudokuBoard.selectedCell;
+    const isSelected =
+      selectedCell == null
+        ? false
+        : c === selectedCell.c && r === selectedCell.r;
+    const conflict = isConflict(r, c, cell);
+    const peer = selectedCell
```

Comment:
Might be more readable something like this:
```
let peer = sameValue = false;
// If a cell has been selected checks if this cell is related to it
if (selectedCell) {
  peer = isCurrentCellPeer(r, c, selectedCell as CellLocation);
  sameValue = doesCurrentCellHaveSameValue(cell);
}
```
Alternatively could just encapsulate all the logic into the helper functions so that they check if a cell has been selected so you are just left with this:
```
peer = isCurrentCellPeer(args)
sameValue = doesCurrentCellHaveSameValue(args)
```

Reply:
I went with the second option you provided as I think it looks cleaner!